pipeline/experiment.py

```diff
@@ -0,0 +1,299 @@
+         ('s1 stim', 8, 'mini-distractors and full distractors (only at late delay), with different levels of the mini-stim and the full-stim during sample period'),
+         ('s1 stim', 9, 'mini-distractors and full distractors (only at late delay), with different levels of the mini-stim and the full-stim during sample period')
+         ]
+
+
+
+@schema
+class Photostim(dj.Manual):
+    definition = """
+    -> Session
+    photo_stim :  smallint 
+    ---
+    -> lab.PhotostimDevice
+    -> BrainLocation
+    ml_location=null: float # um from ref ; right is positive; based on manipulator coordinates/reconstructed track
+    ap_location=null: float # um from ref; anterior is positive; based on manipulator coordinates/reconstructed track
+    dv_location=null: float # um from dura; ventral is positive; based on manipulator coordinates/reconstructed track
+    ml_angle=null: float # Angle between the manipulator/reconstructed track and the Medio-Lateral axis. A tilt towards the right hemishpere is positive.
+    ap_angle=null: float # Angle between the manipulator/reconstructed track and the Anterior-Posterior axis. An anterior tilt is positive.
+    waveform=null:  longblob       # normalized to maximal power. The value of the maximal power is specified for each PhotostimTrialEvent individually
+    frequency=null: float  # (Hz) 
+    """
+
+
+@schema
+class SessionTrial(dj.Imported):
+    definition = """
+    -> Session
+    trial : smallint 		# trial number
+    ---
+    trial_uid=null : int  # unique across sessions/animals
+    start_time : decimal(8, 4)  # (s) relative to session beginning 
+    stop_time=null: decimal(8, 4)  # (s) relative to session beginning 
+    """
+
+
+@schema 
+class TrialNoteType(dj.Lookup):
+    definition = """
+    trial_note_type : varchar(12)
+    """
+    contents = zip(('autolearn', 'protocol #', 'bad', 'bitcode'))
+
+
+@schema
+class TrialNote(dj.Imported):
+    definition = """
+    -> SessionTrial
+    -> TrialNoteType
+    ---
+    trial_note  : varchar(255) 
+    """
+
+
+@schema
+class TrainingType(dj.Lookup):
+    definition = """
+    # Mouse training
+    training_type : varchar(100) # mouse training
+    ---
+    training_type_description : varchar(2000) # description
+    """
+    contents = [
+         ('regular', ''),
+         ('regular + distractor', 'mice were first trained on the regular S1 photostimulation task  without distractors, then the training continued in the presence of distractors'),
+         ('regular or regular + distractor', 'includes both training options')
+         ]
+
+
+@schema
+class SessionTraining(dj.Manual):
+    definition = """
+    -> Session
+    -> TrainingType
+    """
+
+
+@schema
+class SessionTask(dj.Manual):
+    definition = """
+    -> Session
+    -> TaskProtocol
+    """
+
+
+@schema
+class SessionComment(dj.Manual):
+    definition = """
+    -> Session
+    session_comment : varchar(767)
```

Comment:
`session_comment` should probably be a secondary attribute here.

Reply:
Indeed, `session_comment` should be a non-pk attribute. Even if multiple comments are allowed per session, another PK attribute like `comment_id: tinyint` ought to be used.

Reply:
fixed, must be a mistake (forgetting the ---), yet this same error is in MAP still